### ISSUE TITLE
Fancy non-flat indexing

### ISSUE THREAD

**Issue by joschu** [joschu/cgt#11]
Currently, you can do this:

```python

x[0:2,:], x[0:2,1:3]
```

which uses GetSli, and this:

```python

x[ cgt.arange(0,10), cgt.arange(0,20,2) ]
```

which uses GetFlatIndices op after computing the flat indices for the expression.

But you can't do this:

```python

x[ [1,3], : ]
```

We should have an Op that looks like GetSli but takes an integer vector of indices along some axis, say, GetFancySli

**Comment by lopuhin** [comment]
Is it required to implement something like a lookup table commonly used in networks that learn word embeddings? Or there is some workaround?

**Comment by joschu** [comment]
Implemented in a43259d
